Terran-One/cw-simulate-ui
# Refactor MenuDrawer

## Description
Vastly refactor the MenuDrawer component. It is now a submodule rather than a file module & consists of various subcomponents in their own file modules. The original module was already large enough, with a few hundred additional lines incoming in near future changes.

Also introduces `optionsExtras` for a `T1MenuItem` intended for popovers and other menus.

## Test steps

1. with an active simulation, test the menu drawer.
2. Simulation item should navigate to /simulation page.
3. Chains item should navigate to /chains page.
4. Chains item should be expansible/collapsible w/ animated chevron icon.
5. Chains subtree should sync with chains from recoil store.
6. Chains subtree items should navigate to /chains/<chainId> page.
7. Chains item should have a kebab menu which fades in & out when mousing over.
8. Chains options menu should have an item "Add Chain" which opens a placeholder popover.
